src/string_array.c

```diff
@@ -133,6 +134,68 @@ rcutils_string_array_cmp(
   return RCUTILS_RET_OK;
 }
 
+rcutils_ret_t
+rcutils_string_array_resize(
+  rcutils_string_array_t * string_array,
+  size_t new_size)
+{
+  RCUTILS_CHECK_FOR_NULL_WITH_MSG(
+    string_array, "string_array is null", return RCUTILS_RET_INVALID_ARGUMENT);
+
+  if (string_array->size == new_size) {
+    return RCUTILS_RET_OK;
+  }
+
+  rcutils_allocator_t * allocator = &string_array->allocator;
+  RCUTILS_CHECK_ALLOCATOR_WITH_MSG(
+    allocator, "allocator is invalid", return RCUTILS_RET_INVALID_ARGUMENT);
+
+  // Stash entries being removed
+  rcutils_string_array_t to_reclaim = rcutils_get_zero_initialized_string_array();
```

Comment:
Can you elaborate on where that `to_reclaim` array is being used? I understand that we stash the to be deleted entries, but we'll never use that array for anything below. But I might just not see that correctly.

Reply:
The strings are being "moved" to `to_reclaim`, which is `fini`'d when the reallocation is successful. That process frees the contents of the array as well.

If the realloc isn't successful, the copies of the string pointers are `NULL`'d before `to_reclaim` is `fini`'d, so the original contents are untouched.